src/main/java/com/bookstore/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,121 @@
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartService shoppingCartService;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final AuthentificationService authentificationService;
+
+    @Transactional
+    @Override
+    public List<OrderResponseDto> getAll(Pageable pageable) {
+        return orderRepository.findAllByUserId(authentificationService.getUserId()).stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Transactional
+    @Override
+    public OrderResponseDto save(CreateOrderRequestDto requestDto) {
+        User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        ShoppingCart cart = shoppingCartService.getById(user.getId());
+        Order order = new Order();
+        order.setUser(user);
+        order.setShippingAddress(requestDto.getShippingAddress());
+        order.setOrderDate(LocalDateTime.now());
+        order.setTotal(cart.getCartItems().stream()
+                .map(this::getPrice)
+                .reduce(BigDecimal.valueOf(0), BigDecimal::add));
+        orderRepository.save(order);
+        OrderResponseDto dto = orderMapper.toDto(order);
+        dto.setOrderItems(getOrderItemDtos(getOrderItems(cart, order)));
+        return dto;
+    }
+
+    @Transactional
+    @Override
+    public OrderResponseDto updateOrderStatus(Long id, UpdateStatusRequestDto requestDto) {
+        Order order = orderRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(
+                "Can't find order by id: " + id));
+        order.setStatus(requestDto.getStatus());
+        return orderMapper.toDto(orderRepository.save(order));
+    }
+
+    @Transactional
+    @Override
+    public List<OrderItemResponseDto> getAllByOrderId(Long orderId, Pageable pageable) {
+        return getOrderById(orderId).getOrderItems().stream()
+                .map(orderItemMapper::toDto)
+                .toList();
+    }
+
+    @Transactional
+    @Override
+    public OrderItemResponseDto getOrderItemById(Long orderId, Long itemId) {
+        return orderItemRepository.findById(itemId)
+                .map(orderItemMapper::toDto)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can't find order item by id: " + itemId));
+    }
+
+    private Order getOrderById(Long id) {
+        return (Order) orderRepository.findByUserId(authentificationService.getUserId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can't find order by id: " + id));
+    }
+
+    private Set<OrderItem> getOrderItems(ShoppingCart cart, Order order) {
+        return cart.getCartItems().stream()
+                .map(orderItemMapper::mapCartItemToOrderItem)
+                .peek(oi -> saveOrderItem(oi, order))
```

Comment:
```suggestion
                .peek(orderItem -> saveOrderItem(oi, order))
```
it would be easier to read